Mathlib/Data/List/Basic.lean

```diff
@@ -1349,6 +1362,16 @@ theorem take_one_drop_eq_of_lt_length' {l : List α} {n : ℕ} (h : n < l.length
 
 #align list.ext List.ext
 
+theorem ext_get? : ∀ {l₁ l₂ : List α}, (∀ n, l₁.get? n = l₂.get? n) → l₁ = l₂ := List.ext
+
+theorem ext_get?' {l₁ l₂ : List α} (h : l₁.length ≤ l₂.length)
+    (h' : ∀ n < l₂.length, l₁.get? n = l₂.get? n) : l₁ = l₂ := by
```

Comment:
These hypotheses are probably not the most convenient. How about assuming `∀ n < max l₁.length l₂.length, l₁.get? n = l₂.get? n`?

Can you also add `iff` versions for this lemma?

Reply:
This hypothesis was indeed nicer in my applications. I also added iff-versions for the other List ext lemmas, since they seemed to be missing